sn_interface/src/network_knowledge/section_authority_provider.rs

```diff
@@ -325,5 +350,18 @@ pub mod test_utils {
 
             (signed_sap, nodes, secret_key_set.secret_key())
         }
+
+        pub fn _gen_sk_set<R: RngCore>(
+            rng: &mut R,
+            elder_count: usize,
+            sk_threshold_size: Option<usize>,
+        ) -> bls::SecretKeySet {
+            // by default we need sup(elder_count) shares to be valid. Thus threshold should be 1
+            // less than that
+            bls::SecretKeySet::random(
+                sk_threshold_size.unwrap_or(supermajority(elder_count) - 1),
```

Comment:
I think this may underflow when `elder_count` is low. Use a `saturating_sub` here instead. 
```suggestion
                sk_threshold_size.unwrap_or(supermajority(elder_count).saturating_sub(1)),
```

More generally, I always prefer to do saturating subs when working with unsigned ints.